### ISSUE TITLE
Flutter code BoxConstraints errors

### ISSUE THREAD

**Issue by andreahaku** [Supernova-Studio/public-issue-tracker#98]
### Issue
> 	The generated flutter code shows a blank white screen and there are errors related to `BoxConstraints`

### Context
> 	​I tried exporting different Sketch files in project format and they all give the same error related to `BoxConstraints`

```Flutter 1.2.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 8661d8aecd (4 weeks ago) • 2019-02-14 19:19:53 -0800
Engine • revision 3757390fa4
Tools • Dart 2.1.2 (build 2.1.2-dev.0.0 0a7dcf17eb)
```
### Expected result
> 	Screens rendered

### Current result
> 	White screen and the following errors in the console:
```
flutter: The following assertion was thrown during performLayout():     
flutter: BoxConstraints forces an infinite height.                      
flutter: These invalid constraints were provided to RenderPadding's layout() function by the following
flutter: function, which probably computed the invalid constraints in question:
flutter:   RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:738:15)
flutter: The offending constraints were:                                
flutter:   BoxConstraints(0.0<=w<=Infinity, h=Infinity)                 
flutter:                                                                
flutter: When the exception was thrown, this was the stack:             
flutter: #0      BoxConstraints.debugAssertIsValid.<anonymous closure>.throwError (package:flutter/src/rendering/box.dart:504:9)
flutter: #1      BoxConstraints.debugAssertIsValid.<anonymous closure> (package:flutter/src/rendering/box.dart:547:21)
flutter: #2      BoxConstraints.debugAssertIsValid (package:flutter/src/rendering/box.dart:551:6)
flutter: #3      RenderObject.layout (package:flutter/src/rendering/object.dart:1547:24)
flutter: #4      RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:738:15)
flutter: #5      RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #6      RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:738:15)
flutter: #7      RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #8      RenderStack.performLayout (package:flutter/src/rendering/stack.dart:510:15)
flutter: #9      RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #10     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #11     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #12     MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:142:11)
flutter: #13     _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:350:7)
flutter: #14     MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:212:7)
flutter: #15     RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:356:14)
flutter: #16     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #17     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #18     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #19     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #20     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1206:11)
flutter: #21     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #22     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #23     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #24     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #25     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #26     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #27     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #28     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:510:15)
flutter: #29     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #30     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #31     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #32     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #33     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #34     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #35     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #36     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #37     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #38     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #39     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #40     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #41     RenderOffstage.performLayout (package:flutter/src/rendering/proxy_box.dart:3032:13)
flutter: #42     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #43     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:510:15)
flutter: #44     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #45     __RenderTheatre&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #46     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #47     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #48     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #49     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #50     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #51     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #52     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #53     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #54     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #55     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
flutter: #56     RenderObject.layout (package:flutter/src/rendering/object.dart:1632:7)
flutter: #57     RenderView.performLayout (package:flutter/src/rendering/view.dart:151:13)
flutter: #58     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1507:7)
flutter: #59     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:766:18)
flutter: #60     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:329:19)
flutter: #61     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:701:13)
flutter: #62     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:268:5)
flutter: #63     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:988:15)
flutter: #64     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:928:9)
flutter: #65     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:749:7)
flutter: #74     _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
flutter: #75     _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
flutter: #76     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
flutter: (elided 8 frames from package dart:async)                      
flutter:                                                                
flutter: The following RenderObject was being processed when the exception was fired:
flutter:   RenderFlex#cf8b5 relayoutBoundary=up1 NEEDS-LAYOUT NEEDS-PAINT
flutter:   creator: Row ← Column ← Stack ← DecoratedBox ← Container ← MediaQuery ←
flutter:   LayoutId-[<_ScaffoldSlot.body>] ← CustomMultiChildLayout ← AnimatedBuilder ← DefaultTextStyle ←
flutter:   AnimatedDefaultTextStyle ← _InkFeatures-[GlobalKey#97c19 ink renderer] ← ⋯
flutter:   parentData: offset=Offset(0.0, 0.0); flex=null; fit=null (can use size)
flutter:   constraints: BoxConstraints(w=375.0, 0.0<=h<=Infinity)       
flutter:   size: MISSING                                                
flutter:   direction: horizontal                                        
flutter:   mainAxisAlignment: start                                     
flutter:   mainAxisSize: max                                            
flutter:   crossAxisAlignment: stretch                                  
flutter:   textDirection: ltr                                           
flutter:   verticalDirection: down                                      
flutter: This RenderObject had the following descendants (showing up to depth 5):
flutter:   RenderPadding#a8259 NEEDS-LAYOUT NEEDS-PAINT                 
flutter:     RenderConstrainedBox#5d202 NEEDS-LAYOUT NEEDS-PAINT        
flutter:       RenderDecoratedBox#24033 NEEDS-LAYOUT NEEDS-PAINT        
flutter:         RenderFlex#c2f2f NEEDS-LAYOUT NEEDS-PAINT              
flutter:           RenderPositionedBox#4f219 NEEDS-LAYOUT NEEDS-PAINT   
flutter:           RenderFlex#06553 NEEDS-LAYOUT NEEDS-PAINT            
flutter:   RenderFlex#b04a5 NEEDS-LAYOUT NEEDS-PAINT                    
flutter:     RenderPadding#38729 NEEDS-LAYOUT NEEDS-PAINT               
flutter:       RenderConstrainedBox#ff7cb NEEDS-LAYOUT NEEDS-PAINT      
flutter:         RenderSemanticsAnnotations#4d507 NEEDS-LAYOUT NEEDS-PAINT
flutter:           RenderImage#145ba NEEDS-LAYOUT NEEDS-PAINT           
flutter:     RenderPadding#5c2c1 NEEDS-LAYOUT NEEDS-PAINT               
flutter:       RenderConstrainedBox#4a06e NEEDS-LAYOUT NEEDS-PAINT      
flutter:         RenderStack#cb93f NEEDS-LAYOUT NEEDS-PAINT             
flutter:           RenderFlex#faee6 NEEDS-LAYOUT NEEDS-PAINT            
flutter:           RenderFlex#ba722 NEEDS-LAYOUT NEEDS-PAINT            
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
flutter: Another exception was thrown: RenderBox was not laid out: RenderFlex#cf8b5 relayoutBoundary=up1 NEEDS-PAINT
flutter: Another exception was thrown: RenderBox was not laid out: RenderFlex#5179a NEEDS-PAINT
flutter: Another exception was thrown: RenderBox was not laid out: RenderStack#31501 relayoutBoundary=up2 NEEDS-PAINT
flutter: Another exception was thrown: RenderBox was not laid out: RenderDecoratedBox#c283d relayoutBoundary=up1 NEEDS-PAINT
flutter: Another exception was thrown: RenderBox was not laid out: RenderDecoratedBox#c283d relayoutBoundary=up1
flutter: Another exception was thrown: Updated layout information required for RenderOpacity#1aef4 NEEDS-LAYOUT NEEDS-PAINT to calculate semantics.
```

### Possible fix
> 	Go through all the BoxConstraints and fix them manually?

So far now Sketch file has been exported correctly and I've not been able to see the result of it.

Thanks!

**Comment by andreahaku** [comment]
I also tested the "Spacebook" example and the only screen working is the login screen that is the only one with no `BoxConstraints`. Then any button that I tap takes me to screen with them an I have again a white screen with the above errors.

**Comment by artufimtcev** [comment]
This should be resolved in 6.2, please reopen if it still happens, we would appreciate sending us an example with the issue to support@supernova.io
Thanks!
